File: sdk/runanywhere-swift/Sources/RunAnywhere/Public/Configuration/DownloadConfig.swift
import Foundation

/// Download configuration
public struct DownloadConfig {
    /// Maximum concurrent downloads
    public let maxConcurrentDownloads: Int

    /// Number of retry attempts
    public let retryAttempts: Int

    /// Custom cache directory
    public let cacheDirectory: URL?

    /// Download timeout in seconds
    public let timeoutInterval: TimeInterval

    public init(
        maxConcurrentDownloads: Int = 2,
        retryAttempts: Int = 3,
        cacheDirectory: URL? = nil,
        timeoutInterval: TimeInterval = 300
    ) {
        self.maxConcurrentDownloads = maxConcurrentDownloads
        self.retryAttempts = retryAttempts
        self.cacheDirectory = cacheDirectory
        self.timeoutInterval = timeoutInterval
    }
}

File: sdk/runanywhere-android/src/main/java/com/runanywhere/sdk/configuration/DownloadConfig.kt
package com.runanywhere.sdk.configuration

import java.io.File

/**
 * Download configuration
 */
data class DownloadConfig(
    /**
     * Maximum concurrent downloads
     */
    val maxConcurrentDownloads: Int = 2,
    
    /**
     * Number of retry attempts
     */
    val retryAttempts: Int = 3,
    
    /**
     * Custom cache directory
     */
    val cacheDirectory: File? = null,
    
    /**
     * Download timeout in seconds
     */
    val timeoutInterval: Long = 300
) 